invenio_app_ils/document_requests/views.py

```diff
@@ -88,70 +95,163 @@ def __init__(self, serializers, ctx, *args, **kwargs):
+            del record["physical_item_provider"]
 
-class AcceptRequestResource(DocumentRequestActionResource):
+        record.commit()
+        db.session.commit()
+        current_app_ils.document_request_indexer.index(record)
+        return self.make_response(pid, record, 202)
+
+
+class DocumentRequestAcceptResource(DocumentRequestActionResource):
     """Accept document request action resource."""
 
-    view_name = "{}_accept_request"
+    view_name = "{}_accept"
 
     @pass_record
     @need_permissions("document-request-accept")
     def post(self, pid, record, **kwargs):
         """Accept request post method."""
         data = self.loader()
 
-        document_pid = data.get("document_pid")
-        if document_pid:
-            record["document_pid"] = document_pid
+        if record["state"] != "PENDING":
+            raise DocumentRequestError(
+                "You cannot change state to 'ACCEPTED' when the Document "
+                "Request is in state: {}".format(record["state"])
+            )
 
-        physical_item_provider = data.get("physical_item_provider")
-        if physical_item_provider:
-            record["physical_item_provider"] = physical_item_provider
+        if "document_pid" not in record:
+            raise DocumentRequestError(
+                "document_pid is required for the Document Request to be "
+                "accepted."
+            )
+
+        if "physical_item_provider" not in record:
+            raise DocumentRequestError(
+                "physical_item_provider is required for the Document Request "
+                "to be accepted."
+            )
 
         state = data.get("state")
-        if state:
-            record["state"] = state
+        if state != 'ACCEPTED':
+            raise DocumentRequestError(
+                "{} is not a valid state".format(record["state"])
+            )
 
+        record["state"] = state
         record.commit()
         db.session.commit()
         current_app_ils.document_request_indexer.index(record)
         return self.make_response(pid, record, 202)
 
 
-class RejectRequestResource(DocumentRequestActionResource):
+class DocumentRequestRejectResource(DocumentRequestActionResource):
     """Reject request resource."""
 
-    view_name = "{}_reject_request"
+    view_name = "{}_reject"
+
+    def reject_permission_factory(self, record):
+        """Reject permission factory."""
+        return DocumentRequestOwnerPermission(record)
 
     @pass_record
-    @need_permissions("document-request-reject")
+    @need_record_permission("reject_permission_factory")
     def post(self, pid, record, **kwargs):
         """Reject request post method."""
         data = self.loader()
         reject_reason = data.get("reject_reason")
         document_pid = data.get("document_pid")
 
+        if record["state"] != "PENDING":
+            raise DocumentRequestError(
+                "You cannot reject a Document Request that "
```

Comment:
Minor: I would rather show to the user an error msg saying " You cannot cancel..." rather than "You cannot reject", because he clicked on Cancel button. It is the librarian that can reject.
Honestly, I would just change all `reject` to `cancel`, since now we have a proper cancel reason.

Reply:
For the reject -> cancel renaming, does it also apply for the backoffice?